stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -623,10 +633,12 @@ void TaskGroupImpl::offer(AsyncTask *completedTask, AsyncContext *context) {
   // queue when a task polls during next() it will notice that we have a value
   // ready for it, and will process it immediately without suspending.
   assert(!waitQueue.load(std::memory_order_relaxed));
+  SWIFT_TASK_DEBUG_LOG("   // ==== b) enqueue completion ------------------------------------------------", 0);
 
   // Retain the task while it is in the queue;
   // it must remain alive until the task group is alive.
-  swift_retain(completedTask);
+//  SWIFT_TASK_DEBUG_LOG(" enqueue completed, retain completed task = %p", completedTask);
+//  swift_retain(completedTask);
```

Comment:
we basically skip doing any of the manual retains since they are not necessary ever since `fillGroupNextResult` became a thing I believe.

Previously we stored the entire tasks, so the dance was necessary, now the "fill in" moves ownership of the value out of the task